src/synapse.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Mock implementation of the main Synapse class
+ */
+
+import type { Synapse as ISynapse, SynapseOptions, StorageOptions, TokenAmount } from './types.js'
+import { MockStorageService } from './storage-service.js'
+
+export class MockSynapse implements ISynapse {
+  private _options: SynapseOptions
+  private _balance: number = 100 // Mock starting balance
+
+  constructor(options: SynapseOptions) {
+    this._options = options
+    console.log('[Mock] Synapse initialized with options:', {
+      withCDN: options.withCDN,
+      rpcAPI: options.rpcAPI,
+      serviceContract: options.serviceContract,
+    })
+  }
+
+  async balance(): Promise<TokenAmount> {
+    console.log('[Mock] Checking balance...')
+
+    // Simulate network delay
+    await new Promise(resolve => setTimeout(resolve, 200))
+
+    return this._balance
+  }
+
+  async deposit(amount: TokenAmount): Promise<TokenAmount> {
+    const depositAmount = typeof amount === 'string' ? parseFloat(amount) : Number(amount)
```

Comment:
parsing floats will not be great for dealing with amounts, you'd want them to be precise. I suggest (as above) to either use BigInts, or at least strings

Reply:
I think you're suggesting that we only ever deal in full integer token amounts, not fractional amounts? I think you're right, I don't like making the user divide everything by 10^18 but I guess that's common practice in this world.